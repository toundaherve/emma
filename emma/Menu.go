package emma

// MenuService handle menu actions
type MenuService interface {
	MenuOfTheDay() (*Menu, error)
}

// Menu contains available foods
type Menu struct {
	Foods []*Food
}

// Food represents a plate
type Food struct {
	Name string
	Description string
	Price float64
	Image string
}