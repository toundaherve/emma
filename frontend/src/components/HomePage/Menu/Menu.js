import React from "react";

const Menu = () => {
  return (
    <div className="container page-favorite">
      <div className="col12">
        <div className="page-title">
          <h1>Emma's Favorites</h1>
        </div>
      </div>
      <div
        id="chilis-favorite-carousel"
        className="owl-carousel animated fadeIn owl-theme"
        style={{ opacity: "1", display: "block" }}
      >
        <div className="owl-wrapper-outer">
          <div
            className="owl-wrapper"
            style={{ width: "4800px", left: "0px", display: "block" }}
          >
            <div className="owl-item" style={{ width: "400px" }}>
              <div className="guest-section" style={{ cursor: "pointer" }}>
                <div className="favorite-img">
                  <img
                    className="itemImg b-lazy b-loaded"
                    alt="My Chili's Rewards"
                    src="/images/google/xView-My-Favorites.jpg.pagespeed.ic.9AFPkLB4Pw.webp"
                  />
                </div>
                <div className="favorite-desc">
                  <div className="heading-tertiary heading-favorite">
                    LOG IN
                  </div>
                  <div className="outer-border">
                    <div className="inner-border">
                      <div className="extra-info">
                        <a>Log in</a> to your My Chiliâ€™s Rewards account to view
                        your favorites.
                      </div>
                    </div>
                  </div>
                </div>
                <div className="favorite-action">
                  <button
                    className="btn slim add-login-button"
                    data-add-default-btn="LOG IN"
                  >
                    LOG IN
                  </button>
                </div>
              </div>
            </div>
            <div className="owl-item" style={{ width: "400px" }}>
              <div className="favorite-item">
                <form
                  className="js-favorite-form favorite-form"
                  action="/chilis/favorites/add-favorites-to-cart"
                  method="post"
                >
                  <input
                    type="hidden"
                    name="_csrf"
                    value="73f74287-1fe2-4db7-ac06-be59789333fa"
                  />
                  <div className="favorite-img">
                    <input
                      id="item-name-P102596-0"
                      type="hidden"
                      name="chilisFavoriteProducts[0].name"
                      value="Oldtimer with Cheese"
                    />
                    <input
                      id="item-id-P102596-0"
                      type="hidden"
                      name="chilisFavoriteProducts[0].productMasterId"
                      value="P102596"
                    />
                    <input
                      id="item-alt-id-P102596-0"
                      type="hidden"
                      name="chilisFavoriteProducts[0].altProductMasterId"
                      value=""
                    />
                    <input
                      id="item-category-id-P102596-0"
                      type="hidden"
                      name="chilisFavoriteProducts[0].categoryMasterId"
                      value="C1004"
                    />
                    <img
                      className="itemImg b-lazy b-loaded"
                      alt="Oldtimer with Cheese"
                      src="https://images.prismic.io/brinker-chilis/e82db89d793cb44c0df27a29dcbbe218bf3acf51_oldtimerwithcheese.jpg?auto=compress,format"
                      title="Oldtimer with Cheese"
                    />
                  </div>
                  <div className="favorite-desc">
                    <div className="heading-tertiary heading-favorite">
                      Oldtimer with Cheese
                    </div>
                    <div className="outer-border">
                      <div className="inner-border">
                        <div className="extra-info">
                          Beef Patty, No Pink, Fries
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="favorite-action">
                    <button
                      id="P102596-0"
                      className="btn slim add-chilis-button"
                      href="#"
                      data-add-default-btn="Add to order"
                      data-added-to-cart-btn="Added To Cart"
                      data-unable-to-add-btn="Unable To Add "
                    >
                      Add to order
                    </button>
                    <input
                      type="hidden"
                      className="quantity"
                      value=""
                      id="chilisFavoriteProducts0.quantity"
                      name="chilisFavoriteProducts[0].quantity"
                    />
                  </div>
                </form>
              </div>
            </div>
            <div className="owl-item" style={{ width: "400px" }}>
              <div className="favorite-item">
                <form
                  className="js-favorite-form favorite-form"
                  action="/chilis/favorites/add-favorites-to-cart"
                  method="post"
                >
                  <input
                    type="hidden"
                    name="_csrf"
                    value="73f74287-1fe2-4db7-ac06-be59789333fa"
                  />
                  <div className="favorite-img">
                    <input
                      id="item-name-P100026-1"
                      type="hidden"
                      name="chilisFavoriteProducts[1].name"
                      value="Original Full Order Ribs"
                    />
                    <input
                      id="item-id-P100026-1"
                      type="hidden"
                      name="chilisFavoriteProducts[1].productMasterId"
                      value="P100026"
                    />
                    <input
                      id="item-alt-id-P100026-1"
                      type="hidden"
                      name="chilisFavoriteProducts[1].altProductMasterId"
                      value="P108134"
                    />
                    <input
                      id="item-category-id-P100026-1"
                      type="hidden"
                      name="chilisFavoriteProducts[1].categoryMasterId"
                      value="C1002"
                    />
                    <img
                      className="itemImg b-lazy b-loaded"
                      alt="Original Full Order Ribs"
                      src="https://images.prismic.io/brinker-chilis/c34229628023958343a3c99f37c6e58d8814b90a_originalfullorderribs.jpg?auto=compress,format"
                      title="Original Full Order Ribs"
                    />
                  </div>
                  <div className="favorite-desc">
                    <div className="heading-tertiary heading-favorite">
                      Original Full Order Ribs
                    </div>
                    <div className="outer-border">
                      <div className="inner-border">
                        <div className="extra-info">
                          Roasted Street Corn, Fries
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="favorite-action">
                    <button
                      id="P100026-1"
                      className="btn slim add-chilis-button"
                      href="#"
                      data-add-default-btn="Add to order"
                      data-added-to-cart-btn="Added To Cart"
                      data-unable-to-add-btn="Unable To Add "
                    >
                      Add to order
                    </button>
                    <input
                      type="hidden"
                      className="quantity"
                      value=""
                      id="chilisFavoriteProducts1.quantity"
                      name="chilisFavoriteProducts[1].quantity"
                    />
                  </div>
                </form>
              </div>
            </div>
            <div className="owl-item" style={{ width: "400px" }}>
              <div className="favorite-item">
                <form
                  className="js-favorite-form favorite-form"
                  action="/chilis/favorites/add-favorites-to-cart"
                  method="post"
                >
                  <input
                    type="hidden"
                    name="_csrf"
                    value="73f74287-1fe2-4db7-ac06-be59789333fa"
                  />
                  <div className="favorite-img">
                    <input
                      id="item-name-P106433-2"
                      type="hidden"
                      name="chilisFavoriteProducts[2].name"
                      value="Mix &amp; Match Fajitas"
                    />
                    <input
                      id="item-id-P106433-2"
                      type="hidden"
                      name="chilisFavoriteProducts[2].productMasterId"
                      value="P106433"
                    />
                    <input
                      id="item-alt-id-P106433-2"
                      type="hidden"
                      name="chilisFavoriteProducts[2].altProductMasterId"
                      value="P108194"
                    />
                    <input
                      id="item-category-id-P106433-2"
                      type="hidden"
                      name="chilisFavoriteProducts[2].categoryMasterId"
                      value="C1019"
                    />
                    <img
                      className="itemImg b-lazy b-loaded"
                      alt="Mix &amp; Match Fajitas"
                      src="https://images.prismic.io/brinker-chilis/70711607240dd73425c94f38502557da75f73e77_mixnmatchfajitas.jpg?auto=compress,format"
                      title="Mix &amp; Match Fajitas"
                    />
                  </div>
                  <div className="favorite-desc">
                    <div className="heading-tertiary heading-favorite">
                      Mix &amp; Match Fajitas
                    </div>
                    <div className="outer-border">
                      <div className="inner-border">
                        <div className="extra-info">Chicken, Steak</div>
                      </div>
                    </div>
                  </div>
                  <div className="favorite-action">
                    <button
                      id="P106433-2"
                      className="btn slim add-chilis-button"
                      href="#"
                      data-add-default-btn="Add to order"
                      data-added-to-cart-btn="Added To Cart"
                      data-unable-to-add-btn="Unable To Add "
                    >
                      Add to order
                    </button>
                    <input
                      type="hidden"
                      className="quantity"
                      value=""
                      id="chilisFavoriteProducts2.quantity"
                      name="chilisFavoriteProducts[2].quantity"
                    />
                  </div>
                </form>
              </div>
            </div>
            <div className="owl-item" style={{ width: "400px" }}>
              <div className="favorite-item">
                <form
                  className="js-favorite-form favorite-form"
                  action="/chilis/favorites/add-favorites-to-cart"
                  method="post"
                >
                  <input
                    type="hidden"
                    name="_csrf"
                    value="73f74287-1fe2-4db7-ac06-be59789333fa"
                  />
                  <div className="favorite-img">
                    <input
                      id="item-name-P102491-3"
                      type="hidden"
                      name="chilisFavoriteProducts[3].name"
                      value="Crispy Honey-Chipotle Chicken Crispers"
                    />
                    <input
                      id="item-id-P102491-3"
                      type="hidden"
                      name="chilisFavoriteProducts[3].productMasterId"
                      value="P102491"
                    />
                    <input
                      id="item-alt-id-P102491-3"
                      type="hidden"
                      name="chilisFavoriteProducts[3].altProductMasterId"
                      value=""
                    />
                    <input
                      id="item-category-id-P102491-3"
                      type="hidden"
                      name="chilisFavoriteProducts[3].categoryMasterId"
                      value="C1025"
                    />
                    <img
                      className="itemImg b-lazy b-loaded"
                      alt="Crispy Honey-Chipotle Chicken Crispers"
                      src="https://images.prismic.io/brinker-chilis/a18d51e26d756dcb9f5c1fc92c79619580673418_crispyhoneychipotlechickencrispers.jpg?auto=compress,format"
                      title="Crispy Honey-Chipotle Chicken Crispers"
                    />
                  </div>
                  <div className="favorite-desc">
                    <div className="heading-tertiary heading-favorite">
                      Crispy Honey-Chipotle Chicken Crispers
                    </div>
                    <div className="outer-border">
                      <div className="inner-border">
                        <div className="extra-info">Corn On The Cob, Fries</div>
                      </div>
                    </div>
                  </div>
                  <div className="favorite-action">
                    <button
                      id="P102491-3"
                      className="btn slim add-chilis-button"
                      href="#"
                      data-add-default-btn="Add to order"
                      data-added-to-cart-btn="Added To Cart"
                      data-unable-to-add-btn="Unable To Add "
                    >
                      Add to order
                    </button>
                    <input
                      type="hidden"
                      className="quantity"
                      value=""
                      id="chilisFavoriteProducts3.quantity"
                      name="chilisFavoriteProducts[3].quantity"
                    />
                  </div>
                </form>
              </div>
            </div>
            <div className="owl-item" style={{ width: "400px" }}>
              <div className="favorite-item">
                <form
                  className="js-favorite-form favorite-form"
                  action="/chilis/favorites/add-favorites-to-cart"
                  method="post"
                >
                  <input
                    type="hidden"
                    name="_csrf"
                    value="73f74287-1fe2-4db7-ac06-be59789333fa"
                  />
                  <div className="favorite-img">
                    <input
                      id="item-name-P106076-4"
                      type="hidden"
                      name="chilisFavoriteProducts[4].name"
                      value="Smokehouse Combo"
                    />
                    <input
                      id="item-id-P106076-4"
                      type="hidden"
                      name="chilisFavoriteProducts[4].productMasterId"
                      value="P106076"
                    />
                    <input
                      id="item-alt-id-P106076-4"
                      type="hidden"
                      name="chilisFavoriteProducts[4].altProductMasterId"
                      value=""
                    />
                    <input
                      id="item-category-id-P106076-4"
                      type="hidden"
                      name="chilisFavoriteProducts[4].categoryMasterId"
                      value="C1003"
                    />
                    <img
                      className="itemImg b-lazy b-loaded"
                      alt="Smokehouse Combo"
                      src="https://images.prismic.io/brinker-chilis/86e29bbdf8771db825d4d6692b70f75a87c731c7_smokehousecombo.jpg?auto=compress,format"
                      title="Smokehouse Combo"
                    />
                  </div>
                  <div className="favorite-desc">
                    <div className="heading-tertiary heading-favorite">
                      Smokehouse Combo
                    </div>
                    <div className="outer-border">
                      <div className="inner-border">
                        <div className="extra-info">
                          Crispy Honey-Chipotle Chicken CrispersÂ®, Ranch
                          Dressing, Original BBQ Half Order House-Smoked Ribs
                          $2.00, Roasted Street Corn, Fries
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="favorite-action">
                    <button
                      id="P106076-4"
                      className="btn slim add-chilis-button"
                      href="#"
                      data-add-default-btn="Add to order"
                      data-added-to-cart-btn="Added To Cart"
                      data-unable-to-add-btn="Unable To Add "
                    >
                      Add to order
                    </button>
                    <input
                      type="hidden"
                      className="quantity"
                      value=""
                      id="chilisFavoriteProducts4.quantity"
                      name="chilisFavoriteProducts[4].quantity"
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div className="owl-controls">
          <div className="owl-pagination">
            <div className="owl-page active">
              <span className=""></span>
            </div>
            <div className="owl-page">
              <span className=""></span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Menu;
