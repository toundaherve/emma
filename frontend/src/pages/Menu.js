import React, { useState, useRef, useEffect } from "react";
import { connect } from "react-redux";
import {
  Grid,
  Box,
  Typography,
  useTheme,
  Tabs,
  Tab,
  makeStyles,
} from "@material-ui/core";

import Header from "../layouts/Header";
import Footer from "../layouts/Footer";
import Modal from "../layouts/Modal";
import Container from "../components/Container";
import { WithCTACard } from "../components/Card";
import useDeviceType from "../hooks/useDeviceType";
import Popover from "../layouts/Popover";
import Drawer from "../layouts/Drawer";
import Food from "./Food";
import Cart from "./Cart";
import Skeleton from "../components/Skeleton";

import { mockMenu } from "../utils/mockData";
import { setMenu, status } from "../store/menuSlice";

const useStyles = makeStyles((theme) => ({
  tabs: {
    "& .MuiTab-root": {
      fontSize: "16px !important",
      padding: "8px 12px !important",
    },
  },
}));

const categories = {
  mains: "mains",
  entrees: "entrees",
  wines: "wines",
};

const Banner = () => {
  const theme = useTheme();
  const isTablet = useDeviceType("tablet");
  return (
    <Box
      bgcolor={theme.palette.grey[900]}
      marginTop={`${isTablet ? "80px" : "72px"}`}
      height="208px"
      display="flex"
      justifyContent="center"
      alignItems="center"
      color={theme.palette.grey[50]}
    >
      <Typography variant={`${isTablet ? "h4" : "h6"}`}>The Menu</Typography>
    </Box>
  );
};

const Navigation = ({ currentCategory, setCurrentCategory }) => {
  const theme = useTheme();
  const classes = useStyles();

  let activeTab = 0;
  if (currentCategory === categories.main) activeTab = 0;
  if (currentCategory === categories.entrees) activeTab = 1;
  if (currentCategory === categories.wines) activeTab = 2;

  const handleChange = (event, newValue) => {
    switch (newValue) {
      case 0:
        setCurrentCategory(categories.mains);
        break;
      case 1:
        setCurrentCategory(categories.entrees);
        break;
      case 2:
        setCurrentCategory(categories.wines);
        break;

      default:
        setCurrentCategory(categories.main);
        break;
    }
  };

  return (
    <Box border={`1px solid ${theme.palette.grey[300]}`}>
      <Container>
        <Tabs
          value={activeTab}
          onChange={handleChange}
          aria-label="simple tabs example"
          className={classes.tabs}
        >
          <Tab label="Main" />
          <Tab label="EntrÃ©es" />
          <Tab label="Wines" />
        </Tabs>
      </Container>
    </Box>
  );
};

const List = ({ onListItemClick, menu, currentCategory }) => {
  return (
    <Container>
      <Box pt={3} pb={3}>
        {menu.status === status.error ? (
          <Typography variant="body1" align="center">
            Sorry, the menu is not yet available. <br />{" "}
            <strong>Please come back later !</strong>
          </Typography>
        ) : (
          <Grid container spacing={3}>
            {menu[currentCategory].map((item, idx) => (
              <Grid item xs={12} sm={4} key={idx}>
                {menu.status === status.loading && <Skeleton />}

                {menu.status === status.loaded && (
                  <WithCTACard
                    data={item}
                    aspectRatio={315 / 240}
                    ctaText="Add to order"
                    onButtonClick={onListItemClick}
                  />
                )}
              </Grid>
            ))}
          </Grid>
        )}
      </Box>
    </Container>
  );
};

function renderFood(isTablet, isOpenedPopover, closePopover, anchorEl) {
  return isTablet ? (
    <Modal open={isOpenedPopover} onCloseIconClick={closePopover}>
      <Food />
    </Modal>
  ) : (
    <Popover
      open={isOpenedPopover}
      anchorEl={anchorEl}
      onCloseIconClick={closePopover}
      horizontalOrigin="center"
      verticalOrigin="bottom"
    >
      <Food />
    </Popover>
  );
}

function renderShoppingCart(
  isTablet,
  isShoppingCartOpened,
  closeShoppingCart,
  anchorEl
) {
  return isTablet ? (
    <Drawer open={isShoppingCartOpened} onClose={closeShoppingCart}>
      <Cart />
    </Drawer>
  ) : (
    <Popover
      open={isShoppingCartOpened}
      anchorEl={anchorEl}
      onCloseIconClick={closeShoppingCart}
      horizontalOrigin="right"
    >
      <Cart />
    </Popover>
  );
}

function getTheMenu(successHandler, errorHandler) {
  setTimeout(() => {
    successHandler(mockMenu);
  }, 5000);
  // errorHandler();
}

const Menu = ({ menu, setMenu }) => {
  const anchorEl = useRef(null);
  const isTablet = useDeviceType("tablet");

  const [isOpenedPopover, setIsOpenedPopover] = useState(false);
  const [isShoppingCartOpened, setIsShoppingCartOpened] = useState(false);
  const [currentCategory, setCurrentCategory] = useState(categories.mains);

  function openPopover() {
    setIsOpenedPopover(true);
  }

  function closePopover() {
    setIsOpenedPopover(false);
  }

  function openShoppingCart() {
    setIsShoppingCartOpened(true);
  }

  function closeShoppingCart() {
    setIsShoppingCartOpened(false);
  }

  function handleSuccessGettingTheMenu(fetchedMenu) {
    setMenu({ ...fetchedMenu, status: status.loaded });
  }

  function handleErrorGettingTheMenu() {
    setMenu({ ...menu, status: status.error });
  }

  useEffect(() => {
    getTheMenu(handleSuccessGettingTheMenu, handleErrorGettingTheMenu);
  }, []);

  return (
    <div>
      <Header onShoppingCartIconClick={openShoppingCart} />
      <Banner />
      <Navigation
        currentCategory={currentCategory}
        setCurrentCategory={setCurrentCategory}
      />
      <List
        onListItemClick={openPopover}
        menu={menu}
        currentCategory={currentCategory}
      />
      {renderFood(isTablet, isOpenedPopover, closePopover, anchorEl)}
      {renderShoppingCart(
        isTablet,
        isShoppingCartOpened,
        closeShoppingCart,
        anchorEl
      )}
      <Box mt={8}>
        <Footer />
      </Box>
    </div>
  );
};

function mapStateToProps(state) {
  const { menu } = state;
  return { menu };
}

function mapDispatchToProps(dispatch) {
  return {
    setMenu: (menu) => dispatch(setMenu(menu)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
