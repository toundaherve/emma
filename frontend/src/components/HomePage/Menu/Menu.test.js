import React from "react";
import renderer from "react-test-renderer";

import Menu from "./Menu";

it("renders correctly", () => {
  const component = renderer.create(<Menu />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
