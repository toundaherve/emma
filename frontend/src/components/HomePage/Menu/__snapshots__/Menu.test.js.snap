// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container page-favorite"
>
  <div
    class="col12"
  >
    <div
      class="page-title"
    >
      <h1>
        Chili's Favorites
      </h1>
    </div>
  </div>
  <div
    class="owl-carousel animated fadeIn owl-theme"
    id="chilis-favorite-carousel"
    style={
      Object {
        "display": "block",
        "opacity": "1",
      }
    }
  >
    <div
      class="owl-wrapper-outer"
    >
      <div
        class="owl-wrapper"
        style={
          Object {
            "display": "block",
            "left": "0px",
            "width": "4800px",
          }
        }
      >
        <div
          class="owl-item"
          style={
            Object {
              "width": "400px",
            }
          }
        >
          <div
            class="guest-section"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <div
              class="favorite-img"
            >
              <img
                alt="My Chili's Rewards"
                class="itemImg b-lazy b-loaded"
                src="/images/google/xView-My-Favorites.jpg.pagespeed.ic.9AFPkLB4Pw.webp"
              />
            </div>
            <div
              class="favorite-desc"
            >
              <div
                class="heading-tertiary heading-favorite"
              >
                LOG IN
              </div>
              <div
                class="outer-border"
              >
                <div
                  class="inner-border"
                >
                  <div
                    class="extra-info"
                  >
                    <a>
                      Log in
                    </a>
                     to your My Chiliâ€™s Rewards account to view your favorites.
                  </div>
                </div>
              </div>
            </div>
            <div
              class="favorite-action"
            >
              <button
                class="btn slim add-login-button"
                data-add-default-btn="LOG IN"
              >
                LOG IN
              </button>
            </div>
          </div>
        </div>
        <div
          class="owl-item"
          style={
            Object {
              "width": "400px",
            }
          }
        >
          <div
            class="favorite-item"
          >
            <form
              action="/chilis/favorites/add-favorites-to-cart"
              class="js-favorite-form favorite-form"
              method="post"
            >
              <input
                name="_csrf"
                type="hidden"
                value="73f74287-1fe2-4db7-ac06-be59789333fa"
              />
              <div
                class="favorite-img"
              >
                <input
                  id="item-name-P102596-0"
                  name="chilisFavoriteProducts[0].name"
                  type="hidden"
                  value="Oldtimer with Cheese"
                />
                <input
                  id="item-id-P102596-0"
                  name="chilisFavoriteProducts[0].productMasterId"
                  type="hidden"
                  value="P102596"
                />
                <input
                  id="item-alt-id-P102596-0"
                  name="chilisFavoriteProducts[0].altProductMasterId"
                  type="hidden"
                  value=""
                />
                <input
                  id="item-category-id-P102596-0"
                  name="chilisFavoriteProducts[0].categoryMasterId"
                  type="hidden"
                  value="C1004"
                />
                <img
                  alt="Oldtimer with Cheese"
                  class="itemImg b-lazy b-loaded"
                  src="https://images.prismic.io/brinker-chilis/e82db89d793cb44c0df27a29dcbbe218bf3acf51_oldtimerwithcheese.jpg?auto=compress,format"
                  title="Oldtimer with Cheese"
                />
              </div>
              <div
                class="favorite-desc"
              >
                <div
                  class="heading-tertiary heading-favorite"
                >
                  Oldtimer with Cheese
                </div>
                <div
                  class="outer-border"
                >
                  <div
                    class="inner-border"
                  >
                    <div
                      class="extra-info"
                    >
                      Beef Patty, No Pink, Fries
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="favorite-action"
              >
                <button
                  class="btn slim add-chilis-button"
                  data-add-default-btn="Add to order"
                  data-added-to-cart-btn="Added To Cart"
                  data-unable-to-add-btn="Unable To Add "
                  href="#"
                  id="P102596-0"
                >
                  Add to order
                </button>
                <input
                  class="quantity"
                  id="chilisFavoriteProducts0.quantity"
                  name="chilisFavoriteProducts[0].quantity"
                  type="hidden"
                  value=""
                />
              </div>
            </form>
          </div>
        </div>
        <div
          class="owl-item"
          style={
            Object {
              "width": "400px",
            }
          }
        >
          <div
            class="favorite-item"
          >
            <form
              action="/chilis/favorites/add-favorites-to-cart"
              class="js-favorite-form favorite-form"
              method="post"
            >
              <input
                name="_csrf"
                type="hidden"
                value="73f74287-1fe2-4db7-ac06-be59789333fa"
              />
              <div
                class="favorite-img"
              >
                <input
                  id="item-name-P100026-1"
                  name="chilisFavoriteProducts[1].name"
                  type="hidden"
                  value="Original Full Order Ribs"
                />
                <input
                  id="item-id-P100026-1"
                  name="chilisFavoriteProducts[1].productMasterId"
                  type="hidden"
                  value="P100026"
                />
                <input
                  id="item-alt-id-P100026-1"
                  name="chilisFavoriteProducts[1].altProductMasterId"
                  type="hidden"
                  value="P108134"
                />
                <input
                  id="item-category-id-P100026-1"
                  name="chilisFavoriteProducts[1].categoryMasterId"
                  type="hidden"
                  value="C1002"
                />
                <img
                  alt="Original Full Order Ribs"
                  class="itemImg b-lazy b-loaded"
                  src="https://images.prismic.io/brinker-chilis/c34229628023958343a3c99f37c6e58d8814b90a_originalfullorderribs.jpg?auto=compress,format"
                  title="Original Full Order Ribs"
                />
              </div>
              <div
                class="favorite-desc"
              >
                <div
                  class="heading-tertiary heading-favorite"
                >
                  Original Full Order Ribs
                </div>
                <div
                  class="outer-border"
                >
                  <div
                    class="inner-border"
                  >
                    <div
                      class="extra-info"
                    >
                      Roasted Street Corn, Fries
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="favorite-action"
              >
                <button
                  class="btn slim add-chilis-button"
                  data-add-default-btn="Add to order"
                  data-added-to-cart-btn="Added To Cart"
                  data-unable-to-add-btn="Unable To Add "
                  href="#"
                  id="P100026-1"
                >
                  Add to order
                </button>
                <input
                  class="quantity"
                  id="chilisFavoriteProducts1.quantity"
                  name="chilisFavoriteProducts[1].quantity"
                  type="hidden"
                  value=""
                />
              </div>
            </form>
          </div>
        </div>
        <div
          class="owl-item"
          style={
            Object {
              "width": "400px",
            }
          }
        >
          <div
            class="favorite-item"
          >
            <form
              action="/chilis/favorites/add-favorites-to-cart"
              class="js-favorite-form favorite-form"
              method="post"
            >
              <input
                name="_csrf"
                type="hidden"
                value="73f74287-1fe2-4db7-ac06-be59789333fa"
              />
              <div
                class="favorite-img"
              >
                <input
                  id="item-name-P106433-2"
                  name="chilisFavoriteProducts[2].name"
                  type="hidden"
                  value="Mix & Match Fajitas"
                />
                <input
                  id="item-id-P106433-2"
                  name="chilisFavoriteProducts[2].productMasterId"
                  type="hidden"
                  value="P106433"
                />
                <input
                  id="item-alt-id-P106433-2"
                  name="chilisFavoriteProducts[2].altProductMasterId"
                  type="hidden"
                  value="P108194"
                />
                <input
                  id="item-category-id-P106433-2"
                  name="chilisFavoriteProducts[2].categoryMasterId"
                  type="hidden"
                  value="C1019"
                />
                <img
                  alt="Mix & Match Fajitas"
                  class="itemImg b-lazy b-loaded"
                  src="https://images.prismic.io/brinker-chilis/70711607240dd73425c94f38502557da75f73e77_mixnmatchfajitas.jpg?auto=compress,format"
                  title="Mix & Match Fajitas"
                />
              </div>
              <div
                class="favorite-desc"
              >
                <div
                  class="heading-tertiary heading-favorite"
                >
                  Mix & Match Fajitas
                </div>
                <div
                  class="outer-border"
                >
                  <div
                    class="inner-border"
                  >
                    <div
                      class="extra-info"
                    >
                      Chicken, Steak
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="favorite-action"
              >
                <button
                  class="btn slim add-chilis-button"
                  data-add-default-btn="Add to order"
                  data-added-to-cart-btn="Added To Cart"
                  data-unable-to-add-btn="Unable To Add "
                  href="#"
                  id="P106433-2"
                >
                  Add to order
                </button>
                <input
                  class="quantity"
                  id="chilisFavoriteProducts2.quantity"
                  name="chilisFavoriteProducts[2].quantity"
                  type="hidden"
                  value=""
                />
              </div>
            </form>
          </div>
        </div>
        <div
          class="owl-item"
          style={
            Object {
              "width": "400px",
            }
          }
        >
          <div
            class="favorite-item"
          >
            <form
              action="/chilis/favorites/add-favorites-to-cart"
              class="js-favorite-form favorite-form"
              method="post"
            >
              <input
                name="_csrf"
                type="hidden"
                value="73f74287-1fe2-4db7-ac06-be59789333fa"
              />
              <div
                class="favorite-img"
              >
                <input
                  id="item-name-P102491-3"
                  name="chilisFavoriteProducts[3].name"
                  type="hidden"
                  value="Crispy Honey-Chipotle Chicken Crispers"
                />
                <input
                  id="item-id-P102491-3"
                  name="chilisFavoriteProducts[3].productMasterId"
                  type="hidden"
                  value="P102491"
                />
                <input
                  id="item-alt-id-P102491-3"
                  name="chilisFavoriteProducts[3].altProductMasterId"
                  type="hidden"
                  value=""
                />
                <input
                  id="item-category-id-P102491-3"
                  name="chilisFavoriteProducts[3].categoryMasterId"
                  type="hidden"
                  value="C1025"
                />
                <img
                  alt="Crispy Honey-Chipotle Chicken Crispers"
                  class="itemImg b-lazy b-loaded"
                  src="https://images.prismic.io/brinker-chilis/a18d51e26d756dcb9f5c1fc92c79619580673418_crispyhoneychipotlechickencrispers.jpg?auto=compress,format"
                  title="Crispy Honey-Chipotle Chicken Crispers"
                />
              </div>
              <div
                class="favorite-desc"
              >
                <div
                  class="heading-tertiary heading-favorite"
                >
                  Crispy Honey-Chipotle Chicken Crispers
                </div>
                <div
                  class="outer-border"
                >
                  <div
                    class="inner-border"
                  >
                    <div
                      class="extra-info"
                    >
                      Corn On The Cob, Fries
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="favorite-action"
              >
                <button
                  class="btn slim add-chilis-button"
                  data-add-default-btn="Add to order"
                  data-added-to-cart-btn="Added To Cart"
                  data-unable-to-add-btn="Unable To Add "
                  href="#"
                  id="P102491-3"
                >
                  Add to order
                </button>
                <input
                  class="quantity"
                  id="chilisFavoriteProducts3.quantity"
                  name="chilisFavoriteProducts[3].quantity"
                  type="hidden"
                  value=""
                />
              </div>
            </form>
          </div>
        </div>
        <div
          class="owl-item"
          style={
            Object {
              "width": "400px",
            }
          }
        >
          <div
            class="favorite-item"
          >
            <form
              action="/chilis/favorites/add-favorites-to-cart"
              class="js-favorite-form favorite-form"
              method="post"
            >
              <input
                name="_csrf"
                type="hidden"
                value="73f74287-1fe2-4db7-ac06-be59789333fa"
              />
              <div
                class="favorite-img"
              >
                <input
                  id="item-name-P106076-4"
                  name="chilisFavoriteProducts[4].name"
                  type="hidden"
                  value="Smokehouse Combo"
                />
                <input
                  id="item-id-P106076-4"
                  name="chilisFavoriteProducts[4].productMasterId"
                  type="hidden"
                  value="P106076"
                />
                <input
                  id="item-alt-id-P106076-4"
                  name="chilisFavoriteProducts[4].altProductMasterId"
                  type="hidden"
                  value=""
                />
                <input
                  id="item-category-id-P106076-4"
                  name="chilisFavoriteProducts[4].categoryMasterId"
                  type="hidden"
                  value="C1003"
                />
                <img
                  alt="Smokehouse Combo"
                  class="itemImg b-lazy b-loaded"
                  src="https://images.prismic.io/brinker-chilis/86e29bbdf8771db825d4d6692b70f75a87c731c7_smokehousecombo.jpg?auto=compress,format"
                  title="Smokehouse Combo"
                />
              </div>
              <div
                class="favorite-desc"
              >
                <div
                  class="heading-tertiary heading-favorite"
                >
                  Smokehouse Combo
                </div>
                <div
                  class="outer-border"
                >
                  <div
                    class="inner-border"
                  >
                    <div
                      class="extra-info"
                    >
                      Crispy Honey-Chipotle Chicken CrispersÂ®, Ranch Dressing, Original BBQ Half Order House-Smoked Ribs $2.00, Roasted Street Corn, Fries
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="favorite-action"
              >
                <button
                  class="btn slim add-chilis-button"
                  data-add-default-btn="Add to order"
                  data-added-to-cart-btn="Added To Cart"
                  data-unable-to-add-btn="Unable To Add "
                  href="#"
                  id="P106076-4"
                >
                  Add to order
                </button>
                <input
                  class="quantity"
                  id="chilisFavoriteProducts4.quantity"
                  name="chilisFavoriteProducts[4].quantity"
                  type="hidden"
                  value=""
                />
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      class="owl-controls"
    >
      <div
        class="owl-pagination"
      >
        <div
          class="owl-page active"
        >
          <span
            class=""
          />
        </div>
        <div
          class="owl-page"
        >
          <span
            class=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
